plugin/core/windows.py

```diff
@@ -485,6 +491,11 @@ def _handle_message_request(self, params: dict, source: str, client: Client, req
         handler = MessageRequestHandler(self._window.active_view(), client, request_id, params, source)  # type: ignore
         handler.show()
 
+    def restart_session(self, config_name: str) -> None:
+        self._restarting = True
+        self.documents.reset()
```

Comment:
Does that guarantee that only the chosen server and not all servers for the given window are restarted?
(not sure what `reset()` does)

Reply:
I was actually looking at that at well,

https://github.com/sublimelsp/LSP/blob/8800b812a4b23a686059d2d9f11eeb8b5f5fb776/plugin/core/windows.py#L127-L130

https://github.com/sublimelsp/LSP/blob/8800b812a4b23a686059d2d9f11eeb8b5f5fb776/plugin/core/windows.py#L172-L174

`self._documents_states` is just a `set()`

Reply:
So it will reset the `show_definitions` setting to default value and remove active server names from the status. That seems a bit wrong if we are not restarting all of the servers. Remember there can be two or more servers running for specific file.

Reply:
Ahh you are right. Will need to brainstorm on the right approach for this

Reply:
>And re-attaching only happens on opening the file so the file would have to be re-opened for the status to show back.

Well, I am not sure that a re-open is required. `LSP: Restart Servers` does this same logic and the status shows back without a re-open of the file. 

I definitely think we can create a helper function that does something along the lines of

```python
def reset(self, config_name: str) -> None:
    for view in self._window.views():
        if view.settings().get('lsp_language').get(config_name):
            self.detach_view(view)
    self._document_states.clear()
```

So that we only detach the view if it is affected by that language server that is being restarted. Should we still use `self._document_states.clear()` in this case?

Reply:
`self._document_states.clear()` is a set that ends up looking like 

`{'/projects/LSP/plugin/core/windows.py', '/projects/LSP/plugin/core/registry.py'}`

What about changing how this is stored and making it similar to this, then we could clear document_states based on the language server.

```json
{
  "pyls": ["/projects/LSP/plugin/core/windows.py", "/projects/LSP/plugin/core/registry.py"],
  "gopls": ["/projects/LiveSessions/main.go", "/projects/LiveSessions/support.go"]
}
```

We could then call `self._document_states.clear()` or `self._document_states[config_name].clear()`

Reply:
It's a step in the right direction, but I already have plans for this involving the `Session` object. Please keep the changes to a single topic for a pull request :)

Reply:
This was in relation to erase_sessions calling self._document_states.clear()

Is that okay to do even if just restarting a single server?

Reply:
You can go ahead with this PR now.

Reply:
Awesome, will work this on the weekend. 3 days of freedom